lib/routes/routeBackbeat.js

```diff
@@ -637,21 +695,41 @@ const backbeatRoutes = {
 };
 
 function routeBackbeat(clientIP, request, response, log) {
-    log.debug('routing request', { method: 'routeBackbeat' });
-    normalizeBackbeatRequest(request);
+    log.debug('routing request', {
+        method: 'routeBackbeat',
+        url: request.url,
+    });
+    _normalizeBackbeatRequest(request);
     const useMultipleBackend = request.resourceType === 'multiplebackenddata';
-    const invalidRequest = (!request.bucketName ||
-                            !request.objectKey ||
-                            !request.resourceType ||
-                            (!request.query.operation && useMultipleBackend));
-    if (invalidRequest) {
-        log.debug('invalid request', {
+    const invalidRequest =
+          (!request.resourceType ||
+           (_isObjectRequest(request) &&
+            (!request.bucketName || !request.objectKey)) ||
+           (!request.query.operation && useMultipleBackend));
+    const invalidRoute =
+          (backbeatRoutes[request.method] === undefined ||
+           backbeatRoutes[request.method][request.resourceType] === undefined ||
+           (backbeatRoutes[request.method][request.resourceType]
+            [request.query.operation] === undefined &&
+            useMultipleBackend));
```

Comment:
I noticed this behavior in another PR, but in javascript, if the first statement is true, using the `&&` will evaluate and return the value of the second. So in this case, it will always return the value of `useMultipleBackend` if the operation query is undefined.
Not sure if this is expected behavior, just making sure

Reply:
Yes, here we only rely on the boolean value of the expressions being evaluated (`useMultipleBackend` is a boolean)

Reply:
lol sorry I got confused here